sklearn/preprocessing/data.py

```diff
@@ -1719,10 +1726,15 @@ def _transform_selected(X, transform, selected="all", copy=True):
         X_sel = transform(X[:, ind[sel]])
         X_not_sel = X[:, ind[not_sel]]
 
-        if sparse.issparse(X_sel) or sparse.issparse(X_not_sel):
-            return sparse.hstack((X_sel, X_not_sel))
+        if order:
+            # As of now, X is expected to be dense array
```

Comment:
Please check https://github.com/scikit-learn/scikit-learn/pull/7668/files#diff-5ebddebc20987b6125fffc893f5abc4c for my update to this file. Thanks!